app/views/portal/show.rb

```diff
@@ -438,118 +438,84 @@ def content
 
     def named_entity_data
       data = [collect_concept_labels, collect_agent_labels, collect_time_labels, collect_place_labels]
-      present = false
-
-      data.each do |group|
-        if group[:fields].size > 0
-          present = true
-        end
-      end
+      present = data.any? { |group| group[:present] }
       {
         present: present,
         data: data
       }
     end
 
-    def collect_agent_labels
-      fields = []
-      agents = document.fetch('agents', [])
-      (agents).map do |agent|
-        fields << { key: t('site.object.named-entities.who.term'), val: agent[:about], agt_link: true }
-        fields << { key: t('site.object.named-entities.who.label'), vals: normalise_named_entity(agent[:prefLabel]), multi: true }
-      end
+    def named_entity_labels(edm, i18n, *args)
+      fields = document.fetch(edm, []).map do |entity|
+        [:about, :prefLabel] + (args || []).map do |f|
+          named_entity_field_label(entity, f, i18n)
+        end
+      end.flatten.compact
+
       {
-        title: t('site.object.named-entities.who.title'),
+        title: t("site.object.named-entities.#{i18n}.title"),
         fields: fields,
         present: fields.size > 0
       }
     end
 
-    def collect_place_labels
-      fields = []
-      places = document.fetch('places', [])
-      (places).map do |place|
-        fields << { key: t('site.object.named-entities.where.term'), val: place[:about], agt_link: true }
-        fields << { key: t('site.object.named-entities.where.label'), vals: normalise_named_entity(place[:prefLabel]), multi: true }
-        if !place[:latitude].nil?
-          fields << { key: t('site.object.named-entities.where.latitude'), val: place[:latitude] }
-        end
-        if !place[:longitude].nil?
-          fields << { key: t('site.object.named-entities.where.longitude'), val: place[:longitude] }
-        end
+    def named_entity_field_label(entity, field, i18n)
```

Comment:
Cyclomatic complexity for named_entity_field_label is too high. [7/6]<br>Method has too many lines. [12/10]